packages/react-components/react-color-picker-preview/library/src/components/ColorArea/useColorArea.ts

```diff
@@ -9,23 +16,167 @@ import type { ColorAreaProps, ColorAreaState } from './ColorArea.types';
+    const newY = 100 - Math.round(((event.clientY - rect.top) / rect.height) * 100);
+
+    return {
+      x: clamp(newX, MIN, MAX),
+      y: clamp(newY, MIN, MAX),
+    };
+  }
+
+  const abortController = React.useRef(new AbortController()).current;
+
+  React.useEffect(() => {
+    return () => {
+      abortController.abort();
+    };
+  }, [abortController]);
+
+  const dispatchCustomInputChangeEvent = (newColor: HsvColor) => {
+    const nativeInputValueSetter = Object.getOwnPropertyDescriptor(
+      targetWindow.HTMLInputElement.prototype,
+      'value',
+    )!.set;
+    nativeInputValueSetter?.call(yRef.current!, newColor.v.toString());
+    nativeInputValueSetter?.call(xRef.current!, newColor.s.toString());
+
+    const event = new Event('change', { bubbles: true });
+
+    xRef.current?.dispatchEvent(event);
+    yRef.current?.dispatchEvent(event);
+  };
+
+  const requestColorChange = useEventCallback((event: MouseEvent) => {
+    const _coordinates = getCoordinates(event);
+    const newColor = { h: color.h, s: _coordinates.x, v: _coordinates.y, a: 1 };
+
+    dispatchCustomInputChangeEvent(newColor);
+  });
+
+  const handleDocumentMouseMove = React.useCallback(
+    (event: MouseEvent) => {
+      requestColorChange(event);
+    },
+    [requestColorChange],
+  );
+  const handleDocumentMouseUp = useEventCallback(() => {
+    targetDocument?.removeEventListener('mousemove', handleDocumentMouseMove);
+  });
+
+  const _onMouseDown: React.MouseEventHandler<HTMLDivElement> = useEventCallback(event => {
+    event.stopPropagation();
+    event.preventDefault();
+
+    requestColorChange(event.nativeEvent);
+
+    targetDocument?.addEventListener('mousemove', handleDocumentMouseMove, { signal: abortController.signal });
+    targetDocument?.addEventListener('mouseup', handleDocumentMouseUp, { once: true, signal: abortController.signal });
+  });
+
+  const _onChange: React.ChangeEventHandler<HTMLInputElement> = useEventCallback(event => {
+    const newColor = { h: color.h, s: Number(xRef.current!.value), v: Number(yRef.current?.value), a: 1 };
+    setColor(newColor);
+    onChange?.(event, {
+      type: 'change',
+      event,
+      color: newColor,
+    });
+  });
+
+  const _onKeyDown = useEventCallback((event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'ArrowUp' || event.key === 'ArrowDown') {
+      event.preventDefault();
+      const newY = event.key === 'ArrowUp' ? clamp(coordinates.y + 1, MIN, MAX) : clamp(coordinates.y - 1, MIN, MAX);
+      const newColor = { h: color.h, s: coordinates.x, v: newY, a: 1 };
+
+      dispatchCustomInputChangeEvent(newColor);
+    }
+  });
+
+  const rootVariables = {
+    [colorAreaCSSVars.areaXProgressVar]: `${coordinates.x}%`,
+    [colorAreaCSSVars.areaYProgressVar]: `${coordinates.y}%`,
+    [colorAreaCSSVars.thumbColorVar]: 'transparent',
+    [colorAreaCSSVars.mainColorVar]: `hsl(${color.h}, 100%, 50%)` || INITIAL_COLOR,
```

Comment:
The segment `|| INITIAL_COLOR` can be omitted since the expression `hsl(${color.h}, 100%, 50%)` will always return a truthy value.